ngr_spider/util.py

```diff
@@ -174,6 +175,40 @@ def get_service(service_record: CswServiceRecord) -> Union[Service, ServiceError
 def empty_string_if_none(input_str):
     return input_str if input_str is not None else ""
 
+def retrieve_generic_service(service_record, service_type: str):
+    found = False
+    retries = 3
+    message = ""
+    LOGGER.info(f"{service_record.metadata_id} - {service_record.service_url}")
+    while not found and retries > 0:
+        service_temp = None
+        try:
+            if service_type == "wms":
+                service_temp = WebMapService(service_record.service_url, version="1.3.0")
+            elif service_type == "wfs":
+                service_temp = WebFeatureService(service_record.service_url, version="2.0.0")
+            elif service_type == "wcs":
+                service_temp = WebCoverageService(service_record.service_url, version="1.1.0")
+            elif service_type == "wmts":
+                service_temp = WebMapTileService(service_record.service_url)
+            elif service_type == "atom":
+                pass
+            elif service_type == "oat":
+                service_temp = OGCApiTiles(service_record.service_url)
+            elif service_type == "oaf":
+                service_temp = OGCApiFeatures(service_record.service_url)
+            else:
+                pass
+            # The constructor initiates the http(s) connection
+            # If it doesn't except, we are good to go
+            return service_temp, None
+        except requests.exceptions.HTTPError as e:
+            message = f"md-identifier: {service_record.metadata_id} - {e}"
+        except Exception:
+            message = f"exception while retrieving {service_type} cap for service md-identifier: {service_record.metadata_id}, url: {service_record.service_url}"
+        time.sleep(5)
```

Comment:
dit zou beter een exponential backoff kunnen zijn

Reply:
Dat kan maar er moet wel een harde limiet zijn op het aantal pogingen om de spider te laten crashen. Misschien 2, 4 en dan 8 seconden? Dat is ongeveer dezelfde periode en in de praktijk sneller als alles toch blijkt te werken.

Reply:
het is ook iets wat we later kunnen toevoegen als dat nodig blijkt te zijn